core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/AttributeComponent.scala

```diff
@@ -338,14 +338,50 @@ trait AttributeComponent {
       val toSaveAttrMap = toPrimaryKeyMap(attributesToSave)
       val existingAttrMap = toPrimaryKeyMap(existingAttributes)
 
-      // update attributes which are in both to-save and currently-exists, insert attributes which are in save but not exists
-      val (attrsToUpdateMap, attrsToInsertMap) = toSaveAttrMap.partition { case (k, v) => existingAttrMap.keySet.contains(k) }
-      // delete attributes which currently exist but are not in the attributes to save
-      val attributesToDelete = existingAttrMap.filterKeys(! attrsToUpdateMap.keySet.contains(_)).values
-
-      deleteAttributeRecordsById(attributesToDelete.map(_.id).toSeq) andThen
-        batchInsertAttributes(attrsToInsertMap.values.toSeq) andThen
-        AlterAttributesUsingScratchTableQueries.updateAction(insertFunction(attrsToUpdateMap.values.toSeq))
+      val existingKeys = existingAttrMap.keySet
+
+      // insert attributes which are in save but not exists
+      val attributesToInsert = toSaveAttrMap.filterKeys(! existingKeys.contains(_))
+
+      // delete attributes which are in exists but not save
+      val attributesToDelete = existingAttrMap.filterKeys(! toSaveAttrMap.keySet.contains(_))
+
+      // update attributes which:are in both to-save and currently-exists, but have different values.
+      // note that currently-existing attributes will have a populated id e.g. "1234", but to-save will have an id of "0"
+      // therefore, create a collection of unmarshaled existing attributes to use for comparison, ignoring ids
+      val existingUnmarshalled = existingAttributes.toList.map(unmarshalValue)
```

Comment:
`unmarshalValue` only unmarshals attribute values (string, boolean, number, json) so this will skip over other types like attribute entity references and i _think_ `AttributeNull` which explains both of the unit test failures

Reply:
ah, good catch. I can address that - this should be fixable!